File: control-plane/src/test/resources/codebase-memory/samples/java/UserController.java
package com.example;

import com.kratisai.controlplane.model.NodeType;
import com.kratisai.controlplane.model.RelationType;

import com.example.service.UserService;

public class UserController extends BaseController implements UserServiceInterface {

    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @Override
    public String getUser(String id) {
        return userService.getUser(id);
    }

    public String listUsers() {
        return handleRequest("/users");
    }
}

File: control-plane/src/test/resources/codebase-memory/samples/typescript/UserController.ts
import {UserServiceInterface} from "./UserServiceInterface";

abstract class BaseController {
    handleRequest(path: string): string {
        return `Handled: ${path}`;
    }
}

export class UserController extends BaseController implements UserServiceInterface {
    getUser(id: string): string {
        return `User-${id}`;
    }

    listUsers(): string[] {
        return ['user1', 'user2'];
    }

    processRequest(): string {
        return this.handleRequest('/users');
    }
}
